### PR TITLE
feat/perf: add insert_range and contains to DeleteVector

### PR DESCRIPTION
**Pull request by Baunsgaard**
## Which issue does this PR close?

This is the Rust counterpart of the Java optimization in https://github.com/apache/iceberg/pull/15791.

## What changes are included in this PR?

- Add `DeleteVector::insert_range(start, end)` that delegates to `RoaringTreemap::insert_range` for bulk range insertion
- Add `DeleteVector::contains(pos)` for efficient single-position lookups
- Add tests for single-position, cross-key, three-key spanning, empty, reversed, and large contiguous ranges

## Are these changes tested?

Yes, unit tests covering all edge cases listed above are included in `delete_vector::tests`.

### CODE REVIEW

**Comment by Baunsgaard** [comment]
Just to make it complete, here are results from the Rust benchmark. 

Related PRs:
- Java: https://github.com/apache/iceberg/pull/15791
- C++: https://github.com/apache/iceberg-cpp/pull/608
- Rust: https://github.com/apache/iceberg-rust/pull/2292

### Benchmark results

`insert_range()` (native roaring-rs range insert) vs a loop of
individual `insert()` calls (--release, Rust 1.92, 5 iterations after 3
warmup):

| Scenario | Before (insert loop) | After (insert_range) | Speedup |
|---|---|---|---|
| 100 positions | 1.7 µs | 73.6 ns | **23x** |
| 10k positions | 127.4 µs | 76.8 ns | **1,659x** |
| 200k positions | 1.79 ms | 151.4 ns | **11,811x** |
| 1M positions | 12.30 ms | 1.1 µs | **11,196x** |
| 10k cross-key boundary | 194.0 µs | 130.0 ns | **1,492x** |

> The cross-key boundary scenario inserts 10k positions spanning the 32-bit key boundary (e.g., `(1LL << 32) - 5000` to `(1LL << 32) + 5000`). This exercises the key-splitting logic where the range is divided across two internal 32-bit Roaring bitmaps.

File: crates/iceberg/src/delete_vector.rs [hunk 1]
```diff
@@ -45,6 +45,26 @@ impl DeleteVector {
...
+    /// Inserts all positions in the range [start, end) into the delete vector.
+    /// If start == end, this method does nothing and returns 0.
+    ///
+    /// # Panics
+    ///
+    /// Panics if start > end (a reversed range indicates a bug in the caller).
+    ///
+    /// Returns the number of newly inserted positions.
+    #[allow(unused)]
+    pub fn insert_range(&mut self, start: u64, end: u64) -> u64 {
```
Comment (xanderbailey) [inline crates/iceberg/src/delete_vector.rs:57]: Should these be pub if delete_vector is `mod delete_vector;`? Is the idea to make delete_vector public?

**Review by xanderbailey — COMMENTED** [review]
Took a quick look and it seems to make sense to me. I'm wondering where we're expected to use these new paths from?

File: crates/iceberg/src/delete_vector.rs (hunk 1 above)
Comment (Baunsgaard) [reply crates/iceberg/src/delete_vector.rs:57]: Hmm, I do not know, I chose to go with a similar function definition as the insert function.

File: crates/iceberg/src/delete_vector.rs [hunk 2]
```diff
@@ -45,6 +45,26 @@ impl DeleteVector {
...
 
+    /// Inserts all positions in the range [start, end) into the delete vector.
+    /// If start == end, this method does nothing and returns 0.
+    ///
+    /// # Panics
+    ///
+    /// Panics if start > end (a reversed range indicates a bug in the caller).
+    ///
+    /// Returns the number of newly inserted positions.
+    #[allow(unused)]
```
Comment (emkornfield) [inline crates/iceberg/src/delete_vector.rs:56]: Do we expect to be able to use the range function?  How much is the cost of constructing a from a set of  arbitrary rows?

Comment (Baunsgaard) [reply crates/iceberg/src/delete_vector.rs:56]: Good question, I wanted to add the range insert to help in exactly this case!

I already made the followup PR in Java : https://github.com/apache/iceberg/pull/16052

TL;DR; We can get speedups when processing a sorted list of rows with ranges of deleted rows.

**Comment by Baunsgaard** [comment]
Closing for now, will probably come with a more 'thorough' approach later.
